projects/components/src/datagrid/interfaces/datagrid-column.interface.ts

```diff
@@ -40,6 +40,106 @@ export enum GridColumnSortDirection {
  */
 export type FunctionRenderer<T> = (record: T) => string;
 
+/**
+ * A type of button who's displayability does not depend on the selected entity.
+ */
+export interface GlobalButton {
+    /**
+     * The translated text of the button.
+     */
+    label: string;
+    /**
+     * The function that is called when the button is pressed.
+     */
+    handler: () => void;
+    /**
+     * The function that is called to determine if the button should be displayed.
+     */
+    shouldDisplay: () => boolean;
+    /**
+     * The css class the button should have.
+     */
+    class?: string;
+}
+
+/**
+ * A type of button who's displayability dependends on the selected entity.
+ */
+export interface ContextualButton<R> {
+    /**
+     * The translated text of the button.
+     */
+    label: string;
+    /**
+     * The function that is called when the button is pressed.
+     *
+     * @param entity the currently selected entities.
+     */
+    handler: (entity: R[]) => void;
+    /**
+     * The function that is called to determine if the button should be displayed.
```

Comment:
This may be shown as disabled instead of hidden. The intended behavior is disabled, we may need a switch if we want this to be configurable. Let's chat with Martin

Reply:
Updated as per the comment with Martin in the most recent commit. Tell me what you think!